File: svnrepo/RIT_JBF_source/src/IEFFrame.java
/*
 * Copyright (C) 2013 Rochester Institute of Technology
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * See the GNU General Public License for more details.
 */

/**
 * IEFFrame.java
 *
 * This class encapsulates all the functionality required to pop up a
 * frame and display the range and proteins of an IEFProtein.
 *
 * @author Jill Zapoticznyj
 *
 * This code is based off of the ProteinFrame class
 * @author Adam Bazinet - author of ProteinFrame
 */

import java.awt.Frame;
import java.awt.Font;
import java.awt.Rectangle;
import java.awt.Label;
import java.awt.Color;
import java.awt.Dimension;
import java.awt.event.WindowAdapter;
import java.awt.event.WindowEvent;
import java.util.Vector;
import javax.swing.JPanel;
import javax.swing.JScrollPane;

public class IEFFrame extends Frame {

    private Electro2D electro2D;          //reference to calling applet
    private String ptTruncated = "";           //name truncated
    private String names = "";
    private JPanel IEFPanel;           //panel to add components to
    private Rectangle dimensions;         //dimension holders
    private Label titleLabel;   //holds protein name
    private IEFProtein ief;    // the IEFProtein whose information will
                               // be displayed
    private Font theFont;                 //font used in this panel
    private String maxRange;   //the maximum pI value held in the IEFProtein
    private String minRange;   //the minimum pI value held in the IEFProtein
    private JScrollPane scroll; //the pane that allows the user to scroll to
                                //view all of the contents
    private Vector labels;     // the labels storing the protein names
    private static int xlocation = 0;  //the x location of this frame
    private static int ylocation = 0;  // the y location of this frame

    /**
     * Constructor for IEFFrame
     *
     * @param i - the IEFProtein being represented
     */
    public IEFFrame( IEFProtein i ) {
	labels = new Vector();
	
	//set the font for the information being displayed
	theFont = new Font("Arial", Font.PLAIN, 12);
	
	//set ief to the value passed as a parameter to this method
	ief = i;
	
	//set the max and min pI values stored in ief
	maxRange = Double.toString( ief.getMaxPI() );
	minRange = Double.toString( ief.getMinPI() );
	
	//set the title to display the maxRange and minRange
	setTitle( minRange + " - " + maxRange );

	IEFPanel = new JPanel();                  //init panel
	IEFPanel.setLayout(null);                //abs. positioning
	IEFPanel.setBackground(new Color(202,225,255));
	this.addWindowListener(new WindowAdapter() { //allow user to close win.
		public void windowClosing( WindowEvent e ) {
		    dispose();
		}
	    });
	

	this.setBounds(xlocation,ylocation,415,500);   //set frame position
	dimensions = this.getBounds();       //store frame size

	//set the size of IEFPanel and let it know how far to scroll
	IEFPanel.setBounds(0,0,dimensions.width,dimensions.height);
	IEFPanel.scrollRectToVisible( dimensions );

	//get the names of the proteins stored in ief
	Vector v = i.getNames();
      
	int location = 30;
	for( int j = 0; j < v.size(); j++ ){
	    //create a label for each protein in ief and store it in labels
	    labels.add( new Label( (String)v.elementAt( j ) ) );

	    //set the sizes of each label
	    ((Label)labels.elementAt( j ) ).setBounds( 5, location, 390, 20 );
	    ((Label)labels.elementAt( j ) ).setFont( theFont );
	    location = location + 15;
	    IEFPanel.add( (Label)labels.elementAt( j ) );
	}
	
	//adjusts the size of the panel to reflect the number of labels
	// that it holds
	if( dimensions.height < location ){
	    IEFPanel.setPreferredSize( new Dimension( dimensions.width, location) );
	    dimensions = IEFPanel.getBounds();
	    IEFPanel.scrollRectToVisible( dimensions );
	}
	else{
	    IEFPanel.setBounds( 0, 0, dimensions.width, dimensions.height );
	}

	//stores the panel in a scrollable panel
	scroll = new JScrollPane( IEFPanel );
	scroll.setPreferredSize( new Dimension( 415, 500 ) );
	scroll.setWheelScrollingEnabled( true );

	
	this.add(scroll);              //add components
	
	// increment the locations for the next Frame created
	xlocation = xlocation + 10;
	ylocation = ylocation + 20;
	
	// once the location reaches a certain point, start layering
	// frames on top of old frames
	if( ylocation > 500 ){
	    ylocation = 0;
	}
	if( xlocation > 500 ){
	    xlocation = 5;
	}
    }

//	public IEFFrame(IEFProteinSwingVersion elementAt) {
//		// temporary  constructor created for fix compiling error 
//	}

}

File: site/swingjs/j2s/main/java/Electro2D/IEFFrame.js
(function(){var P$=Clazz.newPackage("main.java.Electro2D"),I$=[[0,'java.awt.Font','javax.swing.JPanel','java.awt.Color','java.awt.event.WindowAdapter','java.awt.Label','java.awt.Dimension','javax.swing.JScrollPane']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "IEFFrame", null, 'java.awt.Frame');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['S',['maxRange','minRange'],'O',['IEFPanel','javax.swing.JPanel','dimensions','java.awt.Rectangle','ief','main.java.Electro2D.IEFProtein','theFont','java.awt.Font','scroll','javax.swing.JScrollPane']]
,['I',['xlocation','ylocation']]]

Clazz.newMeth(C$, 'c$$main_java_Electro2D_IEFProtein',  function (i) {
Clazz.super_(C$, this);
this.theFont=Clazz.new_($I$(1,1).c$$S$I$I,["Arial", 0, 12]);
this.ief=i;
this.maxRange=Double.toString$D(this.ief.getMaxPI$());
this.minRange=Double.toString$D(this.ief.getMinPI$());
this.setTitle$S(this.minRange + " - " + this.maxRange );
this.IEFPanel=Clazz.new_($I$(2,1));
this.IEFPanel.setLayout$java_awt_LayoutManager(null);
this.IEFPanel.setBackground$java_awt_Color(Clazz.new_($I$(3,1).c$$I$I$I,[202, 225, 255]));
this.addWindowListener$java_awt_event_WindowListener(((P$.IEFFrame$1||
(function(){/*a*/var C$=Clazz.newClass(P$, "IEFFrame$1", function(){Clazz.newInstance(this, arguments[0],1,C$);}, Clazz.load('java.awt.event.WindowAdapter'), null, 1);

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

Clazz.newMeth(C$, 'windowClosing$java_awt_event_WindowEvent',  function (e) {
this.b$['java.awt.Window'].dispose$.apply(this.b$['java.awt.Window'], []);
});
})()
), Clazz.new_($I$(4,1),[this, null],P$.IEFFrame$1)));
this.setBounds$I$I$I$I(C$.xlocation, C$.ylocation, 415, 500);
this.dimensions=this.getBounds$();
this.IEFPanel.setBounds$I$I$I$I(0, 0, this.dimensions.width, this.dimensions.height);
this.IEFPanel.scrollRectToVisible$java_awt_Rectangle(this.dimensions);
var v=i.getNames$();
var location=30;
for (var j=0; j < v.size$(); j++) {
var l=Clazz.new_([v.elementAt$I(j)],$I$(5,1).c$$S);
l.setBounds$I$I$I$I(5, location, 390, 20);
l.setFont$java_awt_Font(this.theFont);
location=location + 15;
this.IEFPanel.add$java_awt_Component(l);
}
if (this.dimensions.height < location) {
this.IEFPanel.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(6,1).c$$I$I,[this.dimensions.width, location]));
this.dimensions=this.IEFPanel.getBounds$();
this.IEFPanel.scrollRectToVisible$java_awt_Rectangle(this.dimensions);
} else {
this.IEFPanel.setBounds$I$I$I$I(0, 0, this.dimensions.width, this.dimensions.height);
}this.scroll=Clazz.new_($I$(7,1).c$$java_awt_Component,[this.IEFPanel]);
this.scroll.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(6,1).c$$I$I,[415, 500]));
this.scroll.setWheelScrollingEnabled$Z(true);
this.add$java_awt_Component(this.scroll);
C$.xlocation=C$.xlocation + 10;
C$.ylocation=C$.ylocation + 20;
if (C$.ylocation > 500) {
C$.ylocation=0;
}if (C$.xlocation > 500) {
C$.xlocation=5;
}}, 1);

C$.$static$=function(){C$.$static$=0;
C$.xlocation=0;
C$.ylocation=0;
};

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v4');//Created 2022-03-22 08:41:08 Java2ScriptVisitor version 3.3.1-v4 net.sf.j2s.core.jar version 3.3.1-v4
